### ISSUE TITLE
CA1822 for Dispose method is not reported in the error list

### ISSUE THREAD

**Issue by xtqqczze** [dotnet/roslyn-analyzers#5323]
### Analyzer

**Diagnostic ID**: [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822): `Mark members as static`

### Analyzer source

**SDK**: 6.0.100-preview.6.21355.2

### Describe the bug

A violation of CA1822 is reported at build time but not for live analysis.

### Steps To Reproduce

```csharp
    internal class ClientRemotePowerShell : IDisposable
    {
        protected void Dispose(bool disposing) // CA1822
        {
            if (disposing)
            {
                // inputstream.Dispose();
                // outputstream.Dispose();
                // errorstream.Dispose();
            }
        }
```

### Expected behavior

A diagnostic reported in the error list.

### Actual behavior

No diagnostic reported in the error list, only at build time.

## Additional context

Using Visual Studio 2019 Version 16.11.0 Preview 3.0.

**Comment by xtqqczze** [comment]
It seems questionable whether this diagnostic should be reported for a Dispose method at all.

**Comment by xtqqczze** [comment]
Possibly related: https://github.com/dotnet/roslyn/issues/23588
